### PR TITLE
Add governance and policy documents to repo

### PR DESCRIPTION
**Pull request by shonfeder**
Up to this point, the explicit documents of our policies and governance contacts have been kept in the [GitHub wiki](https://github.com/ocaml/opam-repository/wiki) associated with this repository.

However, in [today's opam repo maintenance
meeting](https://github.com/ocaml/opam-repository/wiki/Meeting-notes#2024-11-28),
we resolved that it would be beneficial to move the documents detailing
the policy and governance of this repository into the repository itself.
This provides the following benefits over the current practice:

- The policy will be visible when the repo is checked out, making it easy
  to consult locally and at the site of the work it governs.
- We can suggest and manage changes to the policy via PRs, which will
  - make it easier to discuss changes while leaving a record of those
    discussions, and
  - ensure that proposed changes are visible to users of the repository,
    and available for their commentary and questions.

This change adds the content of the following Wiki pages:

- https://github.com/ocaml/opam-repository/wiki/Governance
- https://github.com/ocaml/opam-repository/wiki/Policies
- https://github.com/ocaml/opam-repository/wiki/Package-Archiving:-Plan

In addition, this PR adds:

- a minor change to the archiving policy in https://github.com/ocaml/opam-repository/pull/26992/commits/aefd85291bd8d79a0639a1a14965a81d00c0cf9f
- a grammar fix to the `x-` fields in https://github.com/ocaml/opam-repository/pull/26992/commits/b7730b6dc40dd317be298ea67dd01dd8cde6852f

Assuming these changes are accepted, the next steps will be to remove this content from the wiki and update relevant links to point to these documents.

### CODE REVIEW

File: governance/policies/archiving.md
Comment (hannesm) [inline governance/policies/archiving.md:4]: this metadata is no longer needed, I guess... the status is soon in action :) and the last-updated is now provided by git.

**Comment by hannesm** [comment]
looks good to me, thanks a lot!

File: governance/README.md
Comment (mseri) [inline governance/README.md:42]:
```suggestion
- [@kit-ty-kate](https://github.com/kit-ty-kate) (former maintainer)
```

File: governance/policies/README.md
Comment (mseri) [inline governance/policies/README.md:12]:
```suggestion
enumerated in the [archiving policy](./archiving.md#inclusion-criteria).
```
was that correct?

File: governance/policies/archiving.md
Comment (mseri) [inline governance/policies/archiving.md:74]:
```suggestion
    - Meaning: Records the unmet [primary repo criteria](#inclusion-criteria)
```

File: governance/README.md
Comment (shonfeder) [reply governance/README.md:42]: @kit-ty-kate would you prefer to be removed from here to ensure you won't be pinged, or listed as a former maintainer for posterity?

File: governance/policies/README.md
Comment (shonfeder) [reply governance/policies/README.md]: Sorry about that! Nov ene sure how I did that, but sorry for missing it in my pre-PR review.

File: governance/README.md
Comment (kit-ty-kate) [reply governance/README.md:42]: mmh, i'm fine, i prefer to be listed. The original reasoning behind the governance document in the wiki was to list everyone who has any kind of special access to the repository for transparency sake.

**Comment by shonfeder** [comment]
We'll need to fix the CI before this can be merged. It currently has very naive handling for sub directories, and merging this as is will break the entire pipeline.

**Comment by hannesm** [comment]
> We'll need to fix the CI before this can be merged. It currently has very naive handling for sub directories, and merging this as is will break the entire pipeline.

Alternatively you can put multiple files at the top level....

**Comment by shonfeder** [comment]
> Alternatively you can put multiple files at the top level....

Indeed, but that seems like needless clutter due to an overly restrictive check, in my current view. I'm happy to reconsider tho.

File: governance/README.md
Comment (shonfeder) [reply governance/README.md]: Got it, thanks! I made this change.

File: governance/policies/archiving.md
Comment (avsm) [inline governance/policies/archiving.md:22]: It's worth cross referencing to what the maintained distributions are here

File: governance/policies/archiving.md
Comment (avsm) [inline governance/policies/archiving.md:3]: This is going to age quickly. I suggest defining some dates in the future at which point the process is either ratified or discarded. This is similar to how (e.g.) IETF drafts expire if they are not standardised.

File: governance/policies/archiving.md
Comment (avsm) [inline governance/policies/archiving.md:94]:
```suggestion
        - `["(any)"]` the maintainer will maintain every single version
```

File: governance/policies/archiving.md
Comment (avsm) [inline governance/policies/archiving.md:56]: This fieldname is quite the mouthful to keep track of...

File: governance/policies/archiving.md
Comment (avsm) [inline governance/policies/archiving.md:89]: An aside: do we have an OCaml implementation of this, similar to `ocaml-version`?

**Review by avsm — APPROVED** [review]

**Comment by avsm** [comment]
Multiple files at the toplevel works fine, I think. See ocaml/ocaml for another repo that does this.

File: governance/policies/archiving.md
Comment (mseri) [reply governance/policies/archiving.md:56]: This will be done automatically via a custom tool in preparation, we can shorten the name but we don't expect that anybody will ever write it by hand in a opam file. It is also something that will only appear in the opam files in the other repo

File: governance/policies/archiving.md
Comment (mseri) [reply governance/policies/archiving.md:3]: We will have dates very soon in the announcement for the implementation. Then we can update this by saying that the archival policy will be consolidated by that day but that we reserve the rights to modify it to fit with the evolution of the repository. Would that be a reasonable compromise?

File: governance/policies/archiving.md
Comment (mseri) [reply governance/policies/archiving.md:89]: No, not yet. We will be working on that after everything needed for the first two phases, which are independent of this, is ready

File: governance/policies/archiving.md
Comment (shonfeder) [reply governance/policies/archiving.md:3]: I had the impression that we wanted this note to be perpetual, as a general warning that policies are not static, and need updating as circumstances and processes change.

That said, IMO, this is implicit a notice to that effect is not really necessary, so I am not opposed to to adding a deadline or just removing this entirely.

File: governance/policies/archiving.md
Comment (shonfeder) [reply governance/policies/archiving.md:22]: Good point! What is the best reference for this? https://ocaml.org/tools/compiler-release-cycle or are you thinking of referencing the compiler versions in linux distros with active LTS support?

Comment (mseri) [reply governance/policies/archiving.md:22]: So we could have a 4.05 threshold now and in May make another PR to move all packages < 4.08

File: governance/policies/archiving.md
Comment (mseri) [reply governance/policies/archiving.md:3]: Let's remove it altogether then

File: governance/policies/archiving.md
Comment (mseri) [reply governance/policies/archiving.md:22]: Oh no, I had read it wrong, it is 4.08 on 20.04 which goes EOL in May, then no change in the plan is really needed since we were supporting still 4.08 in the first cut

**Comment by shonfeder** [comment]
Thanks, @mseri!

**Pull request merged**
